qcodes/instrument_drivers/yokogawa/GS200.py

```diff
@@ -393,25 +396,25 @@ def _set_output(self, output_level: float) -> None:
         auto_enabled = self.auto_range()
 
         if not auto_enabled:
-            self_range = self._cached_range_value
+            self_range = float(self.range.get_latest())
             if self_range is None:
                 raise RuntimeError("Trying to set output but not in"
                                    " auto mode and range is unknown.")
         else:
-            mode = self._cached_mode
+            mode = self.source_mode.get_latest()
             if mode == "CURR":
                 self_range = 200E-3
             else:
                 self_range = 30
 
         # Check we are not trying to set an out of range value
-        if self._cached_range_value is None or abs(output_level)\
+        if self.range.get_latest() is None or abs(output_level)\
                 > abs(self_range):
             # Check that the range hasn't changed
             if not auto_enabled:
                 # Update range
-                self.range()
-                self_range = self._cached_range_value
+                self.range.get_latest()
```

Comment:
This also replaces a call to the insturment with the cache. Is that safe?

Reply:
Self_range updates to the latest range value of the instrument and prevents setting out of range output value. I tested it with all range modes and source modes on the instrument.

Reply:
`self_range = float(self.range.get_latest())` does not communicate with the instrument. It calls the cache. 

Furthermore the line `self.range.get_latest()` does nothing as it just reads the value from the cache and throws it away

Reply:
Correct